### PR TITLE
updated FacilityUsernameViewSet to check for exisiting user

### PR DESCRIPTION
**Pull request by jindal2209**
### Summary

Create Account - Existing username error isn't displayed as expected
Updated the get_queryset method

### Reviewer guidance

On testing on the local server, on status 200 it is now warning for existing user before going to step 2.

### References

----

### Contributor Checklist

PR process:

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md
Peek 2021-03-13 18-05

### CODE REVIEW

**Comment by jindal2209** [comment]
I am unable to figure out why it is still showing linting error please help @jonboiser .

File: kolibri/core/auth/api.py [hunk 1]
```diff
@@ -269,15 +269,18 @@ class FacilityUsernameViewSet(ValuesViewset):
     values = ("username",)
 
     def get_queryset(self):
-        if valid_app_key_on_request(self.request):
-            # Special case for app context to return usernames for
-            # the list display
-            return FacilityUser.objects.all()
```
Comment (jonboiser) [inline kolibri/core/auth/api.py]: This code branch is required and shouldn't be deleted.

Comment (jindal2209) [reply kolibri/core/auth/api.py]: Please review again, if still not up to the mark please tell what I am doing wrong.

**Comment by jonboiser** [comment]
Hi @jindal2209 , thanks for this PR, but upon reviewing it, the PR does not seem to fundamentally change how the code currently works. What you've done here

https://github.com/learningequality/kolibri/pull/7853/files#diff-ef701f556ae9d26458bb0ceaf7f54ceb36947391a39e1c53fa01243e2ca8e62eR277-R281

is actually already provided by these filters. https://github.com/learningequality/kolibri/pull/7853/files#diff-ef701f556ae9d26458bb0ceaf7f54ceb36947391a39e1c53fa01243e2ca8e62eR263-R265

Were you able to reproduce the issue in #7591 before applying your changes? I believe that if you revert your changes, you will see the app works as expected. In #7591 you will see there are special facility settings that are required before you will see the bug. Which is why I recommended not simply changing the `FacilityUsernameViewset`, but creating a new set to check for username duplicates.

**Comment by jindal2209** [comment]
Earlier the username error was coming after the step 2. The api was returning an empty object. When we go to step 2 from step 1 there was no warning for existing user.

What I tried to do is to show existing user error directly after step1.  

I'll revert the changes and create new set with class name CheckExistingUser and router path as you mentioned in the issue as:
GET /api/usernameexists/jonboiser -> returns 200 or 204 with no payload

**Pull request closed without merging**